spinalcordtoolbox/scripts/sct_label_vertebrae.py

```diff
@@ -250,27 +242,31 @@ def main(argv=None):
         fname_disc = os.path.abspath(arguments.discfile)
     else:
         fname_disc = None
-    if arguments.initz is not None:
-        initz = arguments.initz
-        if len(initz) != 2:
-            raise ValueError('--initz takes two arguments: position in superior-inferior direction, label value')
-    if arguments.initcenter is not None:
-        initcenter = arguments.initcenter
+
+    initz = arguments.initz
+    initcenter = arguments.initcenter
     # if user provided text file, parse and overwrite arguments
     if arguments.initfile is not None:
-        file = open(arguments.initfile, 'r')
-        initfile = ' ' + file.read().replace('\n', '')
-        arg_initfile = initfile.split(' ')
-        for idx_arg, arg in enumerate(arg_initfile):
+        args = open(arguments.initfile).read().split()
+        iterator = iter(args)
+        for arg in iterator:
```

Comment:
I considered this, but I think it ends up being complicated, based on the following points:
- `argparse` already has a mechanism for splicing in options from a file, [`fromfile_prefix_chars`](https://docs.python.org/3/library/argparse.html#fromfile-prefix-chars), but it's based on prefix characters (like `@my_arg_file`) rather than an argument (like `-initfile my_arg_file`).
- Looking at [the implementation](https://github.com/python/cpython/blob/ef54c4df8283349603406a9274580556e7cc4bef/Lib/argparse.py#L2116-L2137), it's in a private function and doesn't seem easy to hook into, say with a custom Action.
- Given that we have some required arguments, I'm not sure we can call `parse_args()` with more arguments without causing errors.
- Technically our parser (as explained in the usage message) only looks for `-initz` and `-initcenter` in the file, and ignore all other arguments.

But these are not obvious points, so thanks for bringing it up. It would have been nice to have less duplication of error-checking, and less custom parsing.

Reply:
I think I like this newer version best? 

Though, the responsible part of me thinks I'm going to be bitten in the butt when argparse changes up its internals sometime in the future. But... hhh, the appeal of the way the new version is organized is so strong. 😅

Reply:
Actually wait, what if we did a compromise of... keeping this new custom action, but calling the new `parse_initz` directly? 

That way, the same errors get thrown each time, and we're not having to reach into private functions?